### PR TITLE
http-api: changed eventsource tool

### PR DESCRIPTION
**Pull request by tylershuster**
Changes the build system for the `http-api` package,  exports the types in an importable manner, and changes from native eventsource to `@microsoft/fetch-event-source`

### CODE REVIEW

**Comment by tacryt-socryp** [comment]
What is the advantage of using the Microsoft event source library?

**Comment by tylershuster** [comment]
It uses `fetch` in streaming mode which supports resuming a connection. This should improve time-to-wake performance. It also precludes the necessity of using a polyfill for both fetch and eventsource on the serverside, because it can take an injected fetch object. Because fetch is being actively developed while eventsource has generally stagnated, @liam-fitzgerald felt this was the best course of action. Merely judging by the difficulty I had getting the eventsource polyfill to compile, I'm inclined to agree, plus the theoretical arguments outlined above.

**Comment by Fang-** [comment]
For prior discussion: #4004 and #4157. The former can probably be closed once this is in.

**Comment by tylershuster** [comment]
Yes, assuming this works as intended, this being implemented is 2 PRs away (after state conversion). I had some problems but I'm realizing that was probably due to trying my to resume a connection that hadn't died!

**Comment by tacryt-socryp** [comment]
@tylershuster Is this implementation of Channel.js going to live alongside the existing one, or is it replacing it as the "official JS way of talking to Urbit"? I'm now convinced that this is good and proper for Landscape, but still like how minimal the other Channel.js is for "official implementation spec" (with few dependencies).

**Comment by tylershuster** [comment]
I think "live alongside the existing one" is the approach we're taking. I like channel.js as a quick and dirty way of connecting to Urbit — the key words being both quick (no dependencies) and dirty (no extra utility). It's a nice single-file solution in a pinch, but I think most people will want to use this library.

File: pkg/npm/http-api/example/browser.js [hunk 1]
```diff
@@ -1,17 +1,32 @@
...
 /*!********************************!*\
   !*** ./src/example/browser.js ***!
   \********************************/
-/*! unknown exports (runtime-defined) */
-/*! runtime requirements:  */
+/***/ (() => {
+
 eval("// import Urbit from '../../dist/browser';\n// window.Urbit = Urbit;\n\n//# sourceURL=webpack://@urbit/http-api/./src/example/browser.js?");
+
+/***/ })
```
Comment (liam-fitzgerald) [inline pkg/npm/http-api/example/browser.js:20]: Can you add this to the `.gitignore`?

File: pkg/npm/http-api/example/node.js [hunk 2]
```diff
@@ -1,17 +1,32 @@
...
   !*** ./src/example/node.js ***!
   \*****************************/
-/*! unknown exports (runtime-defined) */
-/*! runtime requirements:  */
+/***/ (() => {
+
 eval("// import Urbit from '../../dist/index';\n// async function blastOff() {\n//   const airlock = await Urbit.authenticate({\n//     ship: 'zod',\n//     url: 'localhost:8080',\n//     code: 'lidlut-tabwed-pillex-ridrup',\n//     verbose: true\n//   });\n//   airlock.subscribe('chat-view', '/primary');\n// }\n// blastOff();\n\n//# sourceURL=webpack://@urbit/http-api/./src/example/node.js?");
+
+/***/ })
+
```
Comment (liam-fitzgerald) [inline pkg/npm/http-api/example/node.js:21]: likewise with `.gitignore`

**Review by liam-fitzgerald — APPROVED** [review]
LGTM, getting a little bugged by the built examples, but it shouldn't block this

**Comment by tylershuster** [comment]
Understood. This is something I'd like to discuss with @jtobin when we auto-build releases

**Pull request merged**